### PR TITLE
Business requirement deliverable 8.0 fix ACL error in BR

### PR DESCRIPTION
**Pull request by cialuo**
Business requirement deliverable 8.0 fix ACL error in BR
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/xie8899/project/pull/35%23discussion_r56611279%22%2C%20%22https%3A//github.com/xie8899/project/pull/35%23discussion_r56611753%22%2C%20%22https%3A//github.com/xie8899/project/pull/35%23discussion_r56612248%22%2C%20%22https%3A//github.com/xie8899/project/pull/35%23discussion_r56615095%22%2C%20%22https%3A//github.com/xie8899/project/pull/35%23issuecomment-198247091%22%2C%20%22https%3A//github.com/xie8899/project/pull/35%23issuecomment-198248419%22%2C%20%22https%3A//github.com/xie8899/project/pull/35%23discussion_r56624771%22%2C%20%22https%3A//github.com/xie8899/project/pull/35%23issuecomment-198256125%22%2C%20%22https%3A//github.com/xie8899/project/pull/35%23issuecomment-198264885%22%2C%20%22https%3A//github.com/xie8899/project/pull/35%23issuecomment-198271817%22%2C%20%22https%3A//github.com/xie8899/project/pull/35%23issuecomment-198271881%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/xie8899/project/pull/35%23issuecomment-198247091%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40elicoidal%20%40duanyp1991%20%22%2C%20%22created_at%22%3A%20%222016-03-18T08%3A12%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7767792%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cialuo%22%7D%7D%2C%20%7B%22body%22%3A%20%22LGTM%22%2C%20%22created_at%22%3A%20%222016-03-18T08%3A20%3A31Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3785252%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/duanyp1991%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40duanyp1991%20Fixed%22%2C%20%22created_at%22%3A%20%222016-03-18T08%3A36%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7767792%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cialuo%22%7D%7D%2C%20%7B%22body%22%3A%20%22LGTM%22%2C%20%22created_at%22%3A%20%222016-03-18T08%3A59%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3785252%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/duanyp1991%22%7D%7D%2C%20%7B%22body%22%3A%20%22LGTM%22%2C%20%22created_at%22%3A%20%222016-03-18T09%3A22%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7600613%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/elicoidal%22%7D%7D%2C%20%7B%22body%22%3A%20%22pls%20deploy%22%2C%20%22created_at%22%3A%20%222016-03-18T09%3A22%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7600613%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/elicoidal%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20bffc8779629737811f338ade252cf3b4760ad6ff%20business_requirement_deliverable/models/business.py%2015%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/xie8899/project/pull/35%23discussion_r56624771%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22use%3A%5Cr%5Cn%60%60%60python%5Cr%5Cnif%20self.unit_price%20and%20self.qty%3A%5Cr%5Cn%20%20%20%20self.price_total%20%3D%20self.unit_price%20%2A%20self.qty%5Cr%5Cnelse%3A%5Cr%5Cn%20%20%20%20self.price_total%20%3D%200%5Cr%5Cn%60%60%60%22%2C%20%22created_at%22%3A%20%222016-03-18T08%3A27%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3785252%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/duanyp1991%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20business_requirement_deliverable/models/business.py%3AL54-63%22%7D%2C%20%22Pull%204c7312e1b0d614794040b60e141a03038fe08933%20business_requirement_deliverable_cost/models/business.py%2023%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/xie8899/project/pull/35%23discussion_r56611279%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20remove%20the%20ACL%20here%3F%20%5Cr%5CnAdding%20ACL%20in%20forms%20looks%20good%20to%20me%20but%20not%20sure%20we%20should%20remove%20here.%5Cr%5Cn%40duanyp1991%22%2C%20%22created_at%22%3A%20%222016-03-18T04%3A04%3A57Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7600613%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/elicoidal%22%7D%7D%2C%20%7B%22body%22%3A%20%22if%20add%20ACL%20in%20py%20file%2C%20it%20will%20trigger%20the%20write%20permissions%20problem.%22%2C%20%22created_at%22%3A%20%222016-03-18T04%3A17%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7767792%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cialuo%22%7D%7D%2C%20%7B%22body%22%3A%20%22not%20cool%21%20adding%20ACL%20in%20forms%20means%20the%20data%20still%20accessible%20through%20xmlrpc%20%28eg%2C%20erppeek.%29.%20I%27ll%20check%20with%20%40cialuo%20to%20try%20other%20solution.%22%2C%20%22created_at%22%3A%20%222016-03-18T04%3A30%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/3785252%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/duanyp1991%22%7D%7D%2C%20%7B%22body%22%3A%20%22that%20is%20my%20point.%20thanks%22%2C%20%22created_at%22%3A%20%222016-03-18T05%3A39%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7600613%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/elicoidal%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20business_requirement_deliverable_cost/models/business.py%3AL73-98%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 4c7312e1b0d614794040b60e141a03038fe08933 business_requirement_deliverable_cost/models/business.py 23'></a> &nbsp;<b><a href='https://github.com/xie8899/project/pull/35#discussion_r56611279'>File: business_requirement_deliverable_cost/models/business.py:L73-98</a></b>
- <a href='https://github.com/elicoidal'></a> Why remove the ACL here?
  Adding ACL in forms looks good to me but not sure we should remove here.
  @duanyp1991
- <a href='https://github.com/cialuo'></a> if add ACL in py file, it will trigger the write permissions problem.
- <a href='https://github.com/duanyp1991'></a> not cool! adding ACL in forms means the data still accessible through xmlrpc (eg, erppeek.). I'll check with @cialuo to try other solution.
- <a href='https://github.com/elicoidal'></a> that is my point. thanks
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/xie8899/project/pull/35#issuecomment-198247091'>General Comment</a></b>
- <a href='https://github.com/cialuo'></a> @elicoidal @duanyp1991
- <a href='https://github.com/duanyp1991'></a> LGTM
- <a href='https://github.com/cialuo'></a> @duanyp1991 Fixed
- <a href='https://github.com/duanyp1991'></a> LGTM
- <a href='https://github.com/elicoidal'></a> LGTM
- <a href='https://github.com/elicoidal'></a> pls deploy
- [ ] <a href='#crh-comment-Pull bffc8779629737811f338ade252cf3b4760ad6ff business_requirement_deliverable/models/business.py 15'></a> &nbsp;<b><a href='https://github.com/xie8899/project/pull/35#discussion_r56624771'>File: business_requirement_deliverable/models/business.py:L54-63</a></b>
- <a href='https://github.com/duanyp1991'></a> use:

``` python
if self.unit_price and self.qty:
self.price_total = self.unit_price * self.qty
else:
self.price_total = 0
```

<a href='https://www.codereviewhub.com/xie8899/project/pull/35?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/xie8899/project/pull/35?approve=1'></a>&nbsp;<a href='https://github.com/xie8899/project/pull/35'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: business_requirement_deliverable_cost/models/business.py [hunk 1]
```diff
@@ -73,46 +73,25 @@ def product_uom_change(self):
...
-    )
-
-
 class BusinessRequirement(models.Model):
     _inherit = "business.requirement"
 
     total_revenue = fields.Float(
         store=True,
-        groups='business_requirement_deliverable_cost.\
-group_business_requirement_estimation',
```
Comment (elicoidal) [inline business_requirement_deliverable_cost/models/business.py:95]: Why remove the ACL here? 
Adding ACL in forms looks good to me but not sure we should remove here.
@duanyp1991

Comment (cialuo) [inline business_requirement_deliverable_cost/models/business.py:95]: if add ACL in py file, it will trigger the write permissions problem.

Comment (duanyp1991) [inline business_requirement_deliverable_cost/models/business.py:95]: not cool! adding ACL in forms means the data still accessible through xmlrpc (eg, erppeek.). I'll check with @cialuo to try other solution.

Comment (elicoidal) [inline business_requirement_deliverable_cost/models/business.py:95]: that is my point. thanks

**Comment by cialuo** [comment]
@elicoidal @duanyp1991

**Comment by duanyp1991** [comment]
others LGTM

File: business_requirement_deliverable/models/business.py [hunk 2]
```diff
@@ -53,7 +54,9 @@ class BusinessRequirementResource(models.Model):
     @api.one
     @api.depends('unit_price', 'qty')
     def _get_price_total(self):
-        self.price_total = self.unit_price * self.qty
+        self.price_total = 0
+        if self.unit_price and self.qty:
+            self.price_total = self.unit_price * self.qty
```
Comment (duanyp1991) [inline business_requirement_deliverable/models/business.py:58]: use:

``` python
if self.unit_price and self.qty:
    self.price_total = self.unit_price * self.qty
else:
    self.price_total = 0
```

**Comment by cialuo** [comment]
@duanyp1991 Fixed

**Comment by duanyp1991** [comment]
LGTM

**Comment by elicoidal** [comment]
LGTM

**Comment by elicoidal** [comment]
pls deploy

**Pull request merged**